linux/LinuxProcessList.c

```diff
@@ -1165,32 +1164,37 @@ static bool LinuxProcessList_readCmdlineFile(Process* process, openat_arg_t proc
 #endif
    if (amtRead > 0) {
       filename[amtRead] = 0;
-      lp->mergedCommand.maxLen += amtRead;  /* accommodate exe */
-      if (!lp->procExe || !String_eq(filename, lp->procExe)) {
-         free_and_xStrdup(&lp->procExe, filename);
-         lp->procExeLen = amtRead;
+      mc->maxLen += amtRead; /* accommodate exe */
+      if (!process->procExe ||
+         (!process->procExeDeleted && !String_eq(filename, process->procExe)) ||
+         (process->procExeDeleted && !String_startsWith(filename, process->procExe))) {
+         free_and_xStrdup(&process->procExe, filename);
+
          /* exe is guaranteed to contain at least one /, but validate anyway */
          while (amtRead && filename[--amtRead] != '/')
             ;
-         lp->procExeBasenameOffset = amtRead + 1;
-         lp->mergedCommand.exeChanged = true;
+
+         process->procExeBasenameOffset = amtRead + 1;
+         mc->exeChanged = true;
 
          const char* deletedMarker = " (deleted)";
-         if (strlen(lp->procExe) > strlen(deletedMarker)) {
-            lp->procExeDeleted = String_eq(lp->procExe + strlen(lp->procExe) - strlen(deletedMarker), deletedMarker);
+         if (strlen(process->procExe) > strlen(deletedMarker)) {
+            process->procExeDeleted = String_eq(process->procExe + strlen(process->procExe) - strlen(deletedMarker), deletedMarker);
+
+            if (process->procExeDeleted) {
+               process->procExe[strlen(process->procExe) - strlen(deletedMarker)] = '\0';
 
-            if (lp->procExeDeleted && strlen(lp->procExe) - strlen(deletedMarker) == 1 && lp->procExe[0] == '/') {
-               lp->procExeBasenameOffset = 0;
+               if (String_eq(process->procExe, "/"))
```

Comment:
Is this right?  String_eq will be comparing the full procExe string not just the first character.

Reply:
Yes. The full string is `/ (deleted)` when read from procfs and the ` (deleted)` is trimmed off the end of that string in the previous line. Thus only `/` remains at this point.